x/ethbridge/keeper/msg_server.go

```diff
@@ -313,14 +313,42 @@ func (srv msgServer) RescueCeth(goCtx context.Context, msg *types.MsgRescueCeth)
 	}
 
 	if err := srv.Keeper.ProcessRescueCeth(ctx, msg); err != nil {
-		logger.Error("keeper failed to process rescue ceth message.", errorMessageKey, err.Error())
+		logger.Error("keeper failed to process rescue native_token message.", errorMessageKey, err.Error())
 		return nil, err
 	}
-	logger.Info("sifnode emit rescue ceth event.",
+	logger.Info("sifnode emit rescue native_token event.",
 		"CosmosSender", msg.CosmosSender,
 		"CosmosSenderSequence", strconv.FormatUint(account.GetSequence(), 10),
 		"CosmosReceiver", msg.CosmosReceiver,
-		"CethAmount", msg.CethAmount)
+		"NativeTokenAmount", msg.CethAmount)
 
 	return &types.MsgRescueCethResponse{}, nil
 }
+
+func (srv msgServer) SetNativeToken(goCtx context.Context, msg *types.MsgSetNativeToken) (*types.MsgSetNativeTokenResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	logger := srv.Keeper.Logger(ctx)
+
+	cosmosSender, err := sdk.AccAddressFromBech32(msg.CosmosSender)
+	if err != nil {
+		return nil, err
+	}
+
+	account := srv.Keeper.accountKeeper.GetAccount(ctx, cosmosSender)
+	if account == nil {
+		logger.Error("account is nil.", "CosmosSender", msg.CosmosSender)
+		return nil, sdkerrors.Wrap(sdkerrors.ErrInvalidAddress, msg.CosmosSender)
+	}
+
+	if err := srv.Keeper.ProcessSetNativeToken(ctx, msg); err != nil {
+		logger.Error("keeper failed to process rescue native_token message.", errorMessageKey, err.Error())
+		return nil, err
+	}
+	logger.Info("sifnode emit rescue native_token event.",
```

Comment:
We should emit an event here so that we can pull this data out later if need be.

Reply:
will add an event.